be/src/exec/spill/spiller.hpp

```diff
@@ -296,6 +308,26 @@ Status PartitionedSpillerWriter::flush(RuntimeState* state, bool is_final_flush,
     DCHECK_EQ(_running_flush_tasks, 0);
     _running_flush_tasks++;
 
+    {
+        // finalize all mem tables of spilling_partitions and allocate blocks
+        for (auto partition : spilling_partitions) {
```

Comment:
MemTable::is_full seems not enough to determine if  it should be flushed. because serde operation would take a quite a long time, it should yield if it spent the time beyond the threshold(e.g. 20ms), so a robust way is to make the flush method restartable and yieldable.

Reply:
> MemTable::is_full seems not enough to determine if it should be flushed. because serde operation would take a quite a long time, it should yield if it spent the time beyond the threshold(e.g. 20ms), so a robust way is to make the flush method restartable and yieldable.

The flush interface is processed in the pipeline thread. If there is an active yield here, the pipeline's scheduling framework cannot handle it.
I think this problem can be solved in other ways. Considering that flush is only possible every time push_chunk is called, if a single call here takes a long time, we can adjust the choose partition strategy to select as few as possible each time. The partition triggers flush, which can make each execution time smoother through amortization. Here I still need to do some testing to decide whether I need to add this strategy.
As for active yield, there is no need to do it here. We can use the existing scheduling mechanism of the pipeline driver.